test/apps/carto-quadbin-heatmap/app.tsx

```diff
@@ -0,0 +1,142 @@
+import React, {useState} from 'react';
+import {createRoot} from 'react-dom/client';
+import {Map} from 'react-map-gl';
+import maplibregl from 'maplibre-gl';
+
+import {PickingInfo} from '@deck.gl/core';
+import {fetchMap, getPaletteGradient, QuadbinHeatmapTileLayer} from '@deck.gl/carto';
+import DeckGL from '@deck.gl/react';
+
+import RangeInput from './range-input';
+
+const PALETTES = {
+  Prism: 'Prism',
+  'Prism 1': 'Prism-1',
+  'Prism 2': 'Prism-2',
+  Vivid: 'Vivid',
+  Sunset: 'Sunset',
+  Temps: 'Temps'
+};
+
+const INITIAL_VIEW_STATE = {
+  longitude: -74,
+  latitude: 40.7,
+  zoom: 11,
+  maxZoom: 16,
+  pitch: 0,
+  bearing: 0
+};
+
+const MAP_STYLE = 'https://basemaps.cartocdn.com/gl/dark-matter-gl-style/style.json';
+
+function getTooltip(info?: PickingInfo): any {
+  if (!info?.object) return null;
+  return info.object.properties.name;
+}
+
+export default function App({layers, initialViewState = INITIAL_VIEW_STATE, mapStyle = MAP_STYLE}) {
+  const [palette, setPalette] = useState(Object.values(PALETTES)[0]);
+  const [rangeScale, setRangeScale] = useState(300);
+  const [radius, setRadius] = useState(20);
+
+  const linearGradient = getPaletteGradient(palette);
+
+  layers = layers.map((l: any) => {
+    const {layerName} = l.constructor;
+    if (!['QuadbinTileLayer'].includes(layerName)) {
+      return l;
+    }
+
+    return new QuadbinHeatmapTileLayer(l.props, {
+      // Heatmap effect props
+      palette,
+      radius,
```

Comment:
@padawannn it would be good to discuss what API would be best for the integration? In particular, adding defining the palette by a single string makes it easy to build a demo, but may be too restrictive for production use